src/radical/pilot/utils/staging_helper.py

```diff
@@ -0,0 +1,190 @@
+        self._backend.mkdir(tgt, flags)
+
+    def rmdir(self, tgt, flags=None):
+        self._backend.rmdir(tgt, flags)
+
+    def copy(self, src, tgt, flags=None):
+        self._backend.copy(src, tgt, flags)
+
+    def move(self, src, tgt, flags=None):
+        self._backend.move(src, tgt, flags)
+
+    def link(self, src, tgt, flags=None):
+        self._backend.link(src, tgt, flags)
+
+    def delete(self, tgt, flags=None):
+        self._backend.delete(tgt, flags)
+
+    def sh_callout(self, url, cmd):
+        return self._backend.sh_callout(url, cmd)
+
+    def handle_staging_directive(self, sd):
+
+        action  = sd['action']
+        src     = sd['source']
+        tgt     = sd['target']
+        uid     = sd.get('uid', '')
+        flags   = sd.get('flags', 0)
+        prof_id = sd.get('prof_id')   # staging on behalf of this entity
+
+        assert action in [COPY, LINK, MOVE, TRANSFER]
+
+        if self._prof:
+            self._prof.prof('staging_start', uid=prof_id, msg=uid)
+
+        self._log.info('%-10s %s', action, src)
+        self._log.info('%-10s %s', '', tgt)
+
+        self.copy(src, tgt, flags)
+
+        if self._prof:
+            self._prof.prof('staging_stop', uid=prof_id, msg=uid)
+
+
+# ------------------------------------------------------------------------------
+#
+class StagingHelper_Local(object):
+
+    def __init__(self, log):
+        self._log = log
+
+    def mkdir(self, tgt, flags):
+        self._log.debug('mkdir %s', tgt)
+        tgt = ru.Url(tgt).path
+        ru.rec_makedir(tgt)
+
+    def rmdir(self, tgt, flags):
+        tgt = ru.Url(tgt).path
+        os.rmdir(tgt)
+
+    def copy(self, src, tgt, flags):
+        self._log.debug('copy  %s %s', src, tgt)
+        src = ru.Url(src).path
+        tgt = ru.Url(tgt).path
+        self.mkdir(os.path.dirname(tgt), flags)
+        ru.sh_callout('cp -r %s %s' % (src, tgt))
+
+    def move(self, src, tgt, flags):
+        src = ru.Url(src).path
+        tgt = ru.Url(tgt).path
+        self.mkdir(os.path.dirname(tgt), flags)
+        shutil.move(src, tgt)
+
+    def link(self, src, tgt, flags):
+        src = ru.Url(src).path
+        tgt = ru.Url(tgt).path
+        self.mkdir(os.path.dirname(tgt), flags)
+        os.link(src, tgt)
+
+    def delete(self, tgt, flags):
+        tgt = ru.Url(tgt).path
+        try   : os.unlink(tgt)
+        except: pass
+
+    def sh_callout(self, url, cmd):
+        return ru.sh_callout(cmd, shell=True)
+
+
+# ------------------------------------------------------------------------------
+#
+class StagingHelper_SAGA(object):
+
+    try:
+        import saga.filesystem              as _rsfs
+        import radical.saga.utils.pty_shell as _rsup
+        _has_saga = True
+    except:
+        _has_saga = False
+
+    def __init__(self, log):
+        self._log = log
+        if not self._has_saga:
+            raise Exception('SAGA-Python not available')
+
+    def mkdir(self, tgt, flags):
+        assert _has_saga
```

Comment:
`self._has_saga` and other its cases below (within the class)

Reply:
Was fixed already - thanks.